ProductService/src/main/java/com/metaphorce/exam/ecommerce/product/controller/DiscountController.java

```diff
@@ -0,0 +1,138 @@
+package com.metaphorce.exam.ecommerce.product.controller;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.metaphorce.exam.ecommerce.product.dto.DiscountDTO;
+import com.metaphorce.exam.ecommerce.product.mapper.DiscountMapper;
+import com.metaphorce.exam.ecommerce.product.model.impl.Discount;
+import com.metaphorce.exam.ecommerce.product.service.impl.DiscountService;
+
+import jakarta.validation.Valid;
+import lombok.extern.slf4j.Slf4j;
+
+@RequestMapping("/discount")
+@RestController
+@Slf4j
+public class DiscountController {
+	@Autowired
+	private DiscountService service;
+	
+	/**
+	 * Implements the HTTP Post to create a Discount
+	 * @param Discount Data to insert
+	 * @return New Discount create
+	 * @author Juan Chavez
+	 * @since Jun/11/2024 
+	 */
+	@PostMapping
+	public ResponseEntity<DiscountDTO> createDiscount(@Valid @RequestBody DiscountDTO DiscountRequest) {
+		log.info("Discount to create: " + DiscountRequest);
+		ResponseEntity<DiscountDTO> response = null;
+		Discount newDiscount = this.service.create(DiscountMapper.mapper.toEntity(DiscountRequest));
+
+		if(newDiscount != null) {
+			response = ResponseEntity.ok(DiscountMapper.mapper.toDTO(newDiscount));
+			log.info("Discount created: " + newDiscount);
+		} else {
+			response = ResponseEntity.badRequest().build();
+			log.info("Error to create Discount: " + DiscountRequest);
+		}
+
+		return response;
+	}
+
+	/**
+	 * Implements the HTTP Get to find all Discounts
+	 * @return List of Discounts
+	 * @author Juan Chavez
+	 * @since Jun/11/2024 
+	 */
+	@GetMapping
+	public ResponseEntity<List<DiscountDTO>> getAllDiscounts() {
+		log.info("Get All Discounts ...");
+		ResponseEntity<List<DiscountDTO>> response = null;
+		List<Discount> listDiscounts = this.service.getAll();
+
+		if(listDiscounts != null) {
```

Comment:
I don't understand why check if the value is null. It would be better to throw exceptions in the service when it doesn't find a certain resource; also, add a global exception handler.

In the case of not being able to create a resource, I think it's better to let a 500 status be generated because it probably means there's something wrong with the system.